pkg/controller/direct/kms/keyhandle/keyhandle_controller.go

```diff
@@ -168,9 +176,34 @@ func (a *Adapter) Create(ctx context.Context, createOp *directbase.CreateOperati
 
 // Update operation not supported for KeyHandle.
 func (a *Adapter) Update(ctx context.Context, updateOp *directbase.UpdateOperation) error {
-	log := klog.FromContext(ctx).WithName(ctrlName)
-	log.V(2).Info("update operation not supported for KeyHandle resource")
-	return nil
+	log := klog.FromContext(ctx)
+	log.V(2).Info("updating Logging Link", "name", a.id)
+	mapCtx := &direct.MapContext{}
+
+	resource := KMSKeyHandleSpec_ToProto(mapCtx, &a.desired.DeepCopy().Spec)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+	resource.Name = a.id.String()
+
+	paths, err := common.CompareProtoMessage(resource, a.actual, common.BasicDiff)
+	if err != nil {
+		return err
+	}
+	if len(paths) == 0 {
+		log.V(2).Info("no field needs update", "name", a.id)
+		status := &krm.KMSKeyHandleStatus{}
+		status.ObservedState = KMSKeyHandleStatusObservedState_FromProto(mapCtx, a.actual)
+		if mapCtx.Err() != nil {
+			return mapCtx.Err()
+		}
+		externalRef := a.actual.Name
+		status.ExternalRef = &externalRef
+		return updateOp.UpdateStatus(ctx, status, nil)
+	} else {
+		return fmt.Errorf("update operation not supported for resource %v %v",
```

Comment:
Not a blocker, but would be great if we make it explicit in the error message about what field got updated unexpectedly.